storage/src/journal/fixed.rs

```diff
@@ -381,79 +384,70 @@ impl<E: Storage + Metrics, A: Codec<Cfg = ()> + FixedSize> Journal<E, A> {
         Ok(A::decode(&buf[..A::SIZE]).unwrap())
     }
 
-    /// Returns an unordered stream of all items in the journal with position >= `start_pos`.
+    /// Returns an ordered stream of all items in the journal with position >= `start_pos`.
     ///
     /// # Integrity
     ///
     /// If any corrupted data is found, the stream will return an error.
-    ///
-    /// # Concurrency
-    ///
-    /// The `concurrency` parameter controls how many blobs are replayed concurrently. This can
-    /// dramatically speed up the replay process if the underlying storage supports concurrent reads
-    /// across different blobs.
     pub async fn replay(
         &self,
-        concurrency: usize,
         buffer: usize,
         start_pos: u64,
     ) -> Result<impl Stream<Item = Result<(u64, A), Error>> + '_, Error> {
-        assert!(concurrency > 0);
         // Collect all blobs to replay
         let start_blob = start_pos / self.cfg.items_per_blob;
-        let blobs = self
-            .blobs
-            .range(start_blob..)
-            .map(|(index, blob)| (index, blob.clone()))
-            .collect::<Vec<_>>();
+        let blobs = self.blobs.range(start_blob..).collect::<Vec<_>>();
+
+        // Need to gather the sizes outside the closure since flat_map doesn't allow async.
+        let mut sizes = Vec::with_capacity(blobs.len());
+        for b in blobs.iter() {
+            sizes.push(b.1.size().await);
```

Comment:
It may simplify some logic to just create `Read` here (and to just pass them into `stream` as blob).

Reply:
Was hoping to avoid pre-allocating those buffers in case there are a big number of blobs.

Reply:
added a comment to explain this.